### PR TITLE
Add BigCommerce loader for Image Component

### PR DESCRIPTION
**Pull request by lfades**

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| buildDuration | 12.9s | 12.8s | -92ms |
| nodeModulesSize | 91.2 MB | 91.2 MB | ⚠️ +679 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.401 | 2.404 | 0 |
| / avg req/sec | 1041.04 | 1040.09 | ⚠️ -0.95 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.309 | 1.259 | -0.05 |
| /error-in-render avg req/sec | 1909.77 | 1986.47 | +76.7 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| 677f882d2ed8..35e7.js gzip | 11.1 kB | 11.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-23d7afb..be6f.js gzip | 7.34 kB | 7.34 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58.1 kB | 58.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.94 kB | 6.94 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-ccae7c3..dule.js gzip | 6.32 kB | 6.32 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 53 kB | 53 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-89ad9e7..25bb.js gzip | 1.34 kB | 1.34 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.74 kB | 7.74 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-aeb707b..dule.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.39 kB | 5.39 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 651 B | 651 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| index.html gzip | 1 kB | 1 kB | ✓ |
| link.html gzip | 1.01 kB | 1.01 kB | ✓ |
| withRouter.html gzip | 994 B | 994 B | ✓ |
| Overall change | 3.01 kB | 3.01 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| buildDuration | 14.5s | 14.5s | -63ms |
| nodeModulesSize | 91.2 MB | 91.2 MB | ⚠️ +679 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| 677f882d2ed8..35e7.js gzip | 11.1 kB | 11.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-23d7afb..be6f.js gzip | 7.34 kB | 7.34 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58.1 kB | 58.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.94 kB | 6.94 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-ccae7c3..dule.js gzip | 6.32 kB | 6.32 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 53 kB | 53 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-89ad9e7..25bb.js gzip | 1.34 kB | 1.34 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.74 kB | 7.74 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-aeb707b..dule.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.39 kB | 5.39 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 651 B | 651 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| _error.js | 1.06 MB | 1.06 MB | ✓ |
| 404.html | 4.73 kB | 4.73 kB | ✓ |
| hooks.html | 3.92 kB | 3.92 kB | ✓ |
| index.js | 1.06 MB | 1.06 MB | ✓ |
| link.js | 1.1 MB | 1.1 MB | ✓ |
| routerDirect.js | 1.1 MB | 1.1 MB | ✓ |
| withRouter.js | 1.1 MB | 1.1 MB | ✓ |
| Overall change | 5.42 MB | 5.42 MB | ✓ |

</details>

</details>
Commit: f31a7a0d55b551cefdc49a5d56e465188c52f2bd

File: packages/next/client/image.tsx [hunk 1]
```diff
@@ -5,10 +5,11 @@ const loaders = new Map<LoaderKey, (props: LoaderProps) => string>([
   ['imgix', imgixLoader],
   ['cloudinary', cloudinaryLoader],
   ['akamai', akamaiLoader],
+  ['bigCommerce', bigCommerceLoader],
```
Comment (bookernath) [inline packages/next/client/image.tsx:8]: `bigCommerce` looks a little strange... perhaps we can just lowercase the entire word

`bigcommerceLoader`

**Review by bookernath — DISMISSED** [review]
String replacement looks good for the resizing functionality, one comment on formatting of company name.

In the future, we could also support WxH resizing if desired.

File: packages/next/client/image.tsx (hunk 1 above)
Comment (lfades) [reply packages/next/client/image.tsx]: Updated 👍

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| buildDuration | 13.2s | 13.3s | ⚠️ +131ms |
| nodeModulesSize | 91.2 MB | 91.2 MB | ⚠️ +679 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.571 | 2.561 | -0.01 |
| / avg req/sec | 972.23 | 976.16 | +3.93 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.53 | 1.548 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 1634.42 | 1615.14 | ⚠️ -19.28 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| 677f882d2ed8..35e7.js gzip | 11.1 kB | 11.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-23d7afb..be6f.js gzip | 7.34 kB | 7.34 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58.1 kB | 58.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.94 kB | 6.94 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-ccae7c3..dule.js gzip | 6.32 kB | 6.32 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 53 kB | 53 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-89ad9e7..25bb.js gzip | 1.34 kB | 1.34 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.74 kB | 7.74 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-aeb707b..dule.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.39 kB | 5.39 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 651 B | 651 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| index.html gzip | 1 kB | 1 kB | ✓ |
| link.html gzip | 1.01 kB | 1.01 kB | ✓ |
| withRouter.html gzip | 994 B | 994 B | ✓ |
| Overall change | 3.01 kB | 3.01 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| buildDuration | 15.2s | 17.1s | ⚠️ +1.9s |
| nodeModulesSize | 91.2 MB | 91.2 MB | ⚠️ +679 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| 677f882d2ed8..35e7.js gzip | 11.1 kB | 11.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-23d7afb..be6f.js gzip | 7.34 kB | 7.34 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58.1 kB | 58.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.94 kB | 6.94 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-ccae7c3..dule.js gzip | 6.32 kB | 6.32 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 53 kB | 53 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-89ad9e7..25bb.js gzip | 1.34 kB | 1.34 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.74 kB | 7.74 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-aeb707b..dule.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.39 kB | 5.39 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 651 B | 651 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | lfades/next.js bigcommerce-image-loader | Change |
| - | - | - | - |
| _error.js | 1.06 MB | 1.06 MB | ✓ |
| 404.html | 4.73 kB | 4.73 kB | ✓ |
| hooks.html | 3.92 kB | 3.92 kB | ✓ |
| index.js | 1.06 MB | 1.06 MB | ✓ |
| link.js | 1.1 MB | 1.1 MB | ✓ |
| routerDirect.js | 1.1 MB | 1.1 MB | ✓ |
| withRouter.js | 1.1 MB | 1.1 MB | ✓ |
| Overall change | 5.42 MB | 5.42 MB | ✓ |

</details>

</details>
Commit: 33ada7e8c4fb24e95f1889a3d413cfecbe9918ff

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 33ada7e8c4fb24e95f1889a3d413cfecbe9918ff

**test/package-managers-tests/basic-pnpm/test/index.test.js** 
-  > pnpm installs
-  > nextjs builds with pnpm

<details>
<summary>Expand output</summary>

  ● pnpm installs

    Command failed: /home/runner/work/next.js/next.js/node_modules/.bin/pnpm add next react react-dom

      28 | 
      29 | const exec = (cmd) => {
    > 30 |   return execSync(cmd, {
         |          ^
      31 |     env: process.env,
      32 |     shell: true,
      33 |     stdio: 'inherit',

      at exec (package-managers-tests/basic-pnpm/test/index.test.js:30:10)
      at Object.<anonymous> (package-managers-tests/basic-pnpm/test/index.test.js:19:3)

  ● nextjs builds with pnpm

    Command failed: /home/runner/work/next.js/next.js/node_modules/.bin/pnpx next build

      28 | 
      29 | const exec = (cmd) => {
    > 30 |   return execSync(cmd, {
         |          ^
      31 |     env: process.env,
      32 |     shell: true,
      33 |     stdio: 'inherit',

      at exec (package-managers-tests/basic-pnpm/test/index.test.js:30:10)
      at Object.<anonymous> (package-managers-tests/basic-pnpm/test/index.test.js:26:3)

</details>

**Pull request closed without merging**